Common/src/Pollster/TaskHandler.cs

```diff
@@ -463,31 +463,44 @@ await workerStreamHandler_.Pipe.CompleteAsync()
   /// <summary>
   ///   Post processing of the acquired task
   /// </summary>
-  /// <param name="cancellationToken">Token used to cancel the execution of the method</param>
+  /// <param name="readCancellationToken">Token used to cancel the wait for the worker reply</param>
+  /// <param name="postCancellationToken">Token used to cancel the operations after the reply is received</param>
+  /// <param name="errorCancellationToken">Token used to cancel the error management operations</param>
   /// <returns>
   ///   Task representing the asynchronous execution of the method
   /// </returns>
   /// <exception cref="NullReferenceException">wrong order of execution</exception>
-  public async Task PostProcessing(CancellationToken cancellationToken)
+  public async Task PostProcessing(CancellationToken readCancellationToken,
+                                   CancellationToken postCancellationToken,
+                                   CancellationToken errorCancellationToken)
   {
     using var _ = logger_.BeginNamedScope("PostProcessing",
                                           ("taskId", messageHandler_.TaskId));
 
-    if (workerStreamHandler_.Pipe == null || taskData_ == null || agent_ == null)
+    if (workerStreamHandler_.Pipe is null)
```

Comment:
I am not sure about the relevance of these tests. They should have been done before (ex: line 444).
However, we should check that before postprocessing, the processing has been finished.

Reply:
We are checking in every method due to nullable warnings.  I changed it to this format because I had one of them null while passing mocks but did not know which one was missing.

Reply:
OK. Keep it that way untill a full refactor of the TaskHandler : acquisition and processing should be split.